platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/ConfigurationAdminImpl.java

```diff
@@ -743,17 +746,39 @@ public ConfigurationStatus enableManagedServiceFactoryConfiguration(
 
     String enabledFactoryPid =
         StringUtils.removeEnd(disabledFactoryPid, ConfigurationStatus.DISABLED_EXTENSION);
-    properties.put(org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID, enabledFactoryPid);
+    Dictionary<String, Object> enabledProperties =
+        copyConfigProperties(properties, enabledFactoryPid);
+    enabledProperties.put(
+        org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID, enabledFactoryPid);
     Configuration enabledConfiguration =
         configurationAdmin.createFactoryConfiguration(enabledFactoryPid, null);
-    enabledConfiguration.update(properties);
+    enabledConfiguration.update(enabledProperties);
 
     disabledConfig.delete();
 
     return new ConfigurationStatusImpl(
         enabledFactoryPid, enabledConfiguration.getPid(), disabledFactoryPid, servicePid);
   }
 
+  private Dictionary<String, Object> copyConfigProperties(
+      Dictionary<String, Object> properties, String factoryPid) {
+    Dictionary<String, Object> copiedProperties = new Hashtable<>();
+    ObjectClassDefinition objectClassDefinition = getObjectClassDefinition(factoryPid);
+    if (objectClassDefinition == null) {
+      LOGGER.debug(
+          "ObjectClassDefinition not found for factoryPid: {}. Unable to copy properties.",
+          factoryPid);
+      return copiedProperties;
+    }
+    Stream.of(objectClassDefinition)
```

Comment:
This is not a registry specific class but this will be a problem for the registry id as I mentioned above.

Reply:
❓ Could the registry ID be `null` if we configured sources manually or through `etc` and not via the registry? We may have to account for that and only copy if it exists.

Reply:
It would be null if it wasn't created through the registry